cmd/cored/main.go

```diff
@@ -236,205 +229,103 @@ func launchConfiguredCore(ctx context.Context, db pg.DB, conf *config.Config, pr
 		chainlog.Fatalkv(ctx, chainlog.KeyError, err)
 	}
 
-	var generatorSigners []generator.BlockSigner
-	var signBlockHandler func(context.Context, *bc.Block) ([]byte, error)
+	var localSigner *blocksigner.BlockSigner
+	var opts []core.LaunchOption
+
+	// Allow loopback/localhost requests in Developer Edition.
+	opts = append(opts, core.AlternateAuth(authLoopbackInDev))
+	opts = append(opts, core.IndexTransactions(*indexTxs))
+	// Add any configured API request rate limits.
+	if *rpsToken > 0 {
+		opts = append(opts, core.RateLimit(limit.AuthUserID, 2*(*rpsToken), *rpsToken))
+	}
+	if *rpsRemoteAddr > 0 {
+		opts = append(opts, core.RateLimit(limit.AuthUserID, 2*(*rpsRemoteAddr), *rpsRemoteAddr))
+	}
+	// If the Core is configured as a block signer, add the sign-block RPC handler.
+	// TODO(jackson): Should this be handled using the callback passed to
+	// API.Handler like how we add the mockHSM routes?
```

Comment:
I am :-1: on this alternative. Using LaunchOption makes perfect sense for this IMO. In fact, it might also make sense to do the same for mockHSM routes too (noted there as well).